experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/lazyField.spec.ts

```diff
@@ -149,314 +148,780 @@ describe("LazyOptionalField", () => {
 
 			// #endregion
 
-			const field = new LazyOptionalField(
+			// #region OptionalField<Any>
+
+			const anyOptionalField = new LazyOptionalField(
 				context,
 				SchemaBuilder.fieldOptional(Any),
 				cursor,
 				detachedFieldAnchor,
 			);
 
 			// Positive cases
-			assert(field.is(SchemaBuilder.fieldOptional(Any)));
+			assert(anyOptionalField.is(SchemaBuilder.fieldOptional(Any)));
 
 			// Negative cases
-			assert(!field.is(SchemaBuilder.fieldOptional()));
-			assert(!field.is(SchemaBuilder.fieldOptional(booleanLeafSchema)));
-			assert(!field.is(SchemaBuilder.fieldValue(Any)));
-			assert(!field.is(SchemaBuilder.fieldSequence(Any)));
+			assert(!anyOptionalField.is(SchemaBuilder.fieldOptional()));
+			assert(!anyOptionalField.is(SchemaBuilder.fieldOptional(leafDomain.boolean)));
+			assert(!anyOptionalField.is(SchemaBuilder.fieldValue()));
+			assert(!anyOptionalField.is(SchemaBuilder.fieldValue(Any)));
+			assert(!anyOptionalField.is(SchemaBuilder.fieldValue(leafDomain.boolean)));
+			assert(!anyOptionalField.is(SchemaBuilder.fieldSequence()));
+			assert(!anyOptionalField.is(SchemaBuilder.fieldSequence(Any)));
+			assert(!anyOptionalField.is(SchemaBuilder.fieldSequence(leafDomain.boolean)));
 			assert(
-				!field.is(SchemaBuilder.fieldRecursive(FieldKinds.value, recursiveStructSchema)),
+				!anyOptionalField.is(
+					SchemaBuilder.fieldRecursive(FieldKinds.value, recursiveStructSchema),
+				),
 			);
-		});
 
-		it("Primitive", () => {
-			// #region Tree and schema initialization
+			// #endregion
 
-			const builder = new SchemaBuilder("test");
-			const booleanLeafSchema = builder.leaf("bool", ValueSchema.Boolean);
-			const numberLeafSchema = builder.leaf("number", ValueSchema.Number);
-			const recursiveStructSchema = builder.structRecursive("recursiveStruct", {
-				flag: SchemaBuilder.fieldValue(booleanLeafSchema),
-				child: SchemaBuilder.fieldRecursive(
-					FieldKinds.optional,
-					() => recursiveStructSchema,
+			// #region OptionalField<Primitive>
+
+			const primitiveOptionalField = new LazyOptionalField(
+				context,
+				SchemaBuilder.fieldOptional(leafDomain.boolean),
+				cursor,
+				detachedFieldAnchor,
+			);
+
+			assert(primitiveOptionalField.is(SchemaBuilder.fieldOptional(leafDomain.boolean)));
+			assert(!primitiveOptionalField.is(SchemaBuilder.fieldOptional(Any)));
+			assert(primitiveOptionalField.is(SchemaBuilder.fieldOptional(leafDomain.number))); // TODO: this is presumably wrong
```

Comment:
@CraigMacomber this does not seem correct. Thoughts?

Reply:
looks like the bug is that `is` uses `compareSets`, which has optional callbacks for extra items in each set, and got wrong  the behaviour when those callbacks are not provided, it should pass aExtra: () => false; eExtra: () => false

Reply:
I'll go ahead and appy the fix here then, unless you would prefer it be addressed in a separate PR.

Reply:
Actually, while that change fixed a couple of the cases, it actually didn't fix this one.